### PR TITLE
Added don't ask checkbox to crash report permission ask dialog

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/18103

User will not see this dialog anymore after checking this checkbox.

Screen Shot 2021-09-15 at 9 58 34 AM

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
`npm run test brave_browser_tests -- --filter=BraveLocalStatePrefsBrowserTest.DefaultLocalStateTest`

1. Launch browser and disable crash report from settings and re-launch
2. Load brave://inducebrowsercrashforrealz/
3. Re-launch and check crash permission ask dialog is launched
4. Check don't ask again checkbox and click `Block`
5. Re-launch to save preferences
5. Load brave://inducebrowsercrashforrealz/ again
6. Re-launch and check crash permission ask dialog is not launched

### CODE REVIEW

File: browser/ui/views/crash_report_permission_ask_dialog_view.cc [hunk 1]
```diff
@@ -231,6 +237,9 @@ void CrashReportPermissionAskDialogView::OnAcceptButtonClicked() {
 }
 
 void CrashReportPermissionAskDialogView::OnWindowClosing() {
+  g_browser_process->local_state()->SetBoolean(
+      kAskCrashReportPermission, !dont_ask_again_checkbox_->GetChecked());
```
Comment (keur) [inline browser/ui/views/crash_report_permission_ask_dialog_view.cc:241]: I won’t hold up the PR on this, but I don’t love the double negative. The permission could be called kDontAskForCrashReporting defaulted false.

**Review by keur — APPROVED** [review]

File: browser/ui/views/crash_report_permission_ask_dialog_view.cc (hunk 1 above)
Comment (simonhong) [reply browser/ui/views/crash_report_permission_ask_dialog_view.cc]: Yeah, it's much better. Thanks for suggestion!
fixed by https://github.com/brave/brave-core/pull/10093/commits/020fb73d0126c76357ad9310caeb5ef567674f46

Comment (bsclifton) [reply browser/ui/views/crash_report_permission_ask_dialog_view.cc]: Thanks for fixing @simonhong 😄 Always easier to fix now - love seeing consistent usage for naming

**Review by karenkliu — APPROVED** [review]
Looks good!

**Review by bsclifton — APPROVED** [review]

**Pull request merged**